### PR TITLE
Init shiori

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

[Shiori](https://github.com/go-shiori/shiori/), the simple bookmarks manager. I don't know much about the Go programming lanuage, and much less how to package it. Please tell me if there could be improvements.

This PR comes with a module and a test for the module

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/shiori.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+in {
+  options = {
+    services.shiori = {
+      enable = mkEnableOption "Shiori simple bookmarks manager";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.shiori;
+        defaultText = "pkgs.shiori";
+        description = "The Shiori package tu use.";
```
Comment (alexarice) [inline nixos/modules/services/web-apps/shiori.nix]:
```suggestion
        description = "The Shiori package to use.";
```

File: nixos/modules/services/web-apps/shiori.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        type = types.str;
+        default = "";
+        description = ''
+          The IP address on which Shiori will listen.
+          If empty, listens on all interfaces.
+        '';
+      };
+
+      port = mkOption {
+        type = types.ints.u16;
```
Comment (alexarice) [inline nixos/modules/services/web-apps/shiori.nix]: I think there is a port type

File: nixos/modules/services/web-apps/shiori.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      serviceConfig = {
+        ExecStart = "${package}/bin/shiori serve --address '${address}' --port '${toString port}'";
+        DynamicUser = true;
+        Environment = "SHIORI_DIR=/var/lib/shiori";
+        StateDirectory = "shiori";
+      };
+    };
+  };
+
+  meta.maintainers = with lib.maintainers; [ minijackson ];
```
Comment (alexarice) [inline nixos/modules/services/web-apps/shiori.nix]:
```suggestion
  meta.maintainers = with maintainers; [ minijackson ];
```
you already have `with lib;` higher up

File: pkgs/servers/web-apps/shiori/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
```
Comment (alexarice) [inline pkgs/servers/web-apps/shiori/default.nix]: See #69229. I think `buildGoPackage` is preferred, though I also know little about go.

Comment (minijackson) [reply pkgs/servers/web-apps/shiori/default.nix]: I have tried with `buildGoPackage`, but it failed after compilation and tests succeeded :-/

Comment (alexarice) [reply pkgs/servers/web-apps/shiori/default.nix]: I have never used either so cannot really help. I just remembered seeing the pr I linked earlier

Comment (alexarice) [reply pkgs/servers/web-apps/shiori/default.nix]: it seems lots of other prs are using buildGoModule, so I wouldn't worry about it unless someone else suggests changing it

Comment (minijackson) [reply pkgs/servers/web-apps/shiori/default.nix]: Well, the Nixpkgs documentation does say that:

> The function `buildGoPackage` builds legacy Go programs, not supporting Go modules.

This gives the impression that `buildGoModule` is preferred ^^

Comment (alexarice) [reply pkgs/servers/web-apps/shiori/default.nix]: sure does

**Comment by minijackson** [comment]
Thanks for all the feedback, @alexarice !

**Comment by alexarice** [comment]
@GrahamcOfBorg build shiori

**Review by alexarice — APPROVED** [review]
Builds and test passes with nix-review

**Comment by marsam** [comment]
Thanks @minijackson and @alexarice

**Pull request merged**